tests/linearizability/model.go

```diff
@@ -96,21 +95,25 @@ func step(state EtcdState, request etcdRequest, response etcdResponse) (bool, Et
 	if state.Key != request.key {
 		panic("Multiple keys not supported")
 	}
+	if response.revision != 0 && response.revision < state.LastRevision {
+		return false, state
+	}
 	switch request.op {
 	case Get:
 		if state.Value == response.getData {
 			return true, state
 		}
-		for write := range state.FailedWrites {
-			if write == response.getData {
-				state.Value = response.getData
-				delete(state.FailedWrites, write)
-				return true, state
-			}
+		_, ok := state.FailedWrites[response.getData]
+		if ok {
+			state.Value = response.getData
+			state.LastRevision = response.revision
+			delete(state.FailedWrites, response.getData)
+			return true, state
 		}
 	case Put:
 		if response.err == nil {
 			state.Value = request.putData
+			state.LastRevision = response.revision
```

Comment:
We might need to update `state.LastRevision` as long as `response.revision != 0`

Reply:
Fixed by requiring revision on Put to always increase.